napari_builtins/_tests/test_io.py

```diff
@@ -312,28 +310,37 @@ def test_add_zarr(write_spec):
     assert out[0].shape == ZARR1.shape  # type: ignore
 
 
-def test_add_zarr_1d_array_is_ignored():
+def test_add_zarr_1d_array_is_ignored(tmp_path):
+    zarr_dir = str(tmp_path / 'data.zarr')
     # For more details: https://github.com/napari/napari/issues/1471
-    with TemporaryDirectory(suffix='.zarr') as zarr_dir:
-        z = zarr.open(zarr_dir, 'w')
-        z['1d'] = np.zeros(3)
 
-        image_path = os.path.join(zarr_dir, '1d')
-        assert npe2.read([image_path], stack=False) == [(None,)]
+    z = zarr.open(store=zarr_dir, mode='w')
+    z.zeros(name='1d', shape=(3,), chunks=(3,), dtype='float32')
+
+    image_path = os.path.join(zarr_dir, '1d')
+    assert npe2.read([image_path], stack=False) == [(None,)]
 
 
-def test_add_many_zarr_1d_array_is_ignored():
+@pytest.mark.xfail(
+    parse_version(version('zarr')) >= parse_version('3.0.0a0')
+    and os.name == 'nt',
+    reason='zarr 3 return incorrect key in windows',
+    strict=True,
+)
```

Comment:
I need to skip this test on windows because of 

https://github.com/zarr-developers/zarr-python/issues/2438

described here https://napari.zulipchat.com/#narrow/channel/212875-general/topic/handling.20zarr.203.2E0.20release/near/480752197

As it is strict `xfail` it will fail once zarr fix the bug.